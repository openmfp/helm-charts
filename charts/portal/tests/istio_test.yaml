suite: deploy
chart:
  version: 1.0.0
  appVersion: 1.0.0
tests:
  - it: matches the snapshot
    values:
      - ../test-values.yaml
    asserts:
      - matchSnapshot: {}
  - it: matches the snapshot with istio disabled
    values:
      - ../test-values.yaml
    set:
      istio:
        enabled: false
    asserts:
      - matchSnapshot: {}
