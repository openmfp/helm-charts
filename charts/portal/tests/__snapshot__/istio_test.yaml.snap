matches the snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: RELEASE-NAME-portal
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-portal
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: RELEASE-NAME-portal
        spec:
          containers:
            - env:
                - name: HTTP_PROTOCOL
                  value: https
                - name: OIDC_CLIENT_ID_PORTAL
                  value: "12345"
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: DISCOVERY_ENDPOINT_PORTAL
                  value: https://auth.example.com/.well-known/openid-configuration
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: portal
                - name: FEATURE_TOGGLES
                  value: enableSessionAutoRefresh=true
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: 1.0.0
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: VALID_WEBCOMPONENT_URLS
                  value: .?
              image: ghcr.io/openmfp/portal:1.0.0
              imagePullPolicy: IfNotPresent
              name: RELEASE-NAME-portal
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  cpu: "1"
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
          securityContext:
            fsGroup: 2000
            runAsGroup: 3000
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-portal
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      name: RELEASE-NAME-portal-client-secret-portal
      namespace: NAMESPACE
    spec:
      data:
        - remoteRef:
            conversionStrategy: Default
            key: dxp-core-team/manual-secrets/portal-client-secrets
            property: portal
          secretKey: secret
      refreshInterval: 10m
      secretStoreRef:
        kind: SecretStore
        name: environment-store
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: portal-client-secret-portal
  3: |
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      host: RELEASE-NAME-portal.NAMESPACE.svc.cluster.local
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
  4: |
    apiVersion: security.istio.io/v1beta1
    kind: PeerAuthentication
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      mtls:
        mode: STRICT
      selector:
        matchLabels:
          app: RELEASE-NAME-portal
  5: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - portal.example.com
        - portal.portal.example.com
        - portal2.example.com
        - portal.portal2.example.com
      http:
        - corsPolicy:
            allowHeaders:
              - Authorization
              - Content-Type
              - '*'
            allowMethods:
              - GET
              - POST
              - PUT
              - DELETE
          route:
            - destination:
                host: RELEASE-NAME-portal.NAMESPACE.svc.cluster.local
                port:
                  number: 8080
  6: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: contentconfigurations-reader
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - contentconfigurations
        verbs:
          - get
          - watch
          - list
  7: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: read-contentconfigurations-global
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: contentconfigurations-reader
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-portal
        namespace: NAMESPACE
  8: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
  9: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME-portal
      type: ClusterIP
  10: |
    apiVersion: networking.istio.io/v1beta1
    kind: Sidecar
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      outboundTrafficPolicy:
        mode: ALLOW_ANY
      workloadSelector:
        labels:
          app: RELEASE-NAME-portal
matches the snapshot with istio disabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: RELEASE-NAME-portal
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-portal
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: RELEASE-NAME-portal
        spec:
          containers:
            - env:
                - name: HTTP_PROTOCOL
                  value: https
                - name: OIDC_CLIENT_ID_PORTAL
                  value: "12345"
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: DISCOVERY_ENDPOINT_PORTAL
                  value: https://auth.example.com/.well-known/openid-configuration
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: portal
                - name: FEATURE_TOGGLES
                  value: enableSessionAutoRefresh=true
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: 1.0.0
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: VALID_WEBCOMPONENT_URLS
                  value: .?
              image: ghcr.io/openmfp/portal:1.0.0
              imagePullPolicy: IfNotPresent
              name: RELEASE-NAME-portal
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  cpu: "1"
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
          securityContext:
            fsGroup: 2000
            runAsGroup: 3000
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-portal
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      name: RELEASE-NAME-portal-client-secret-portal
      namespace: NAMESPACE
    spec:
      data:
        - remoteRef:
            conversionStrategy: Default
            key: dxp-core-team/manual-secrets/portal-client-secrets
            property: portal
          secretKey: secret
      refreshInterval: 10m
      secretStoreRef:
        kind: SecretStore
        name: environment-store
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: portal-client-secret-portal
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: contentconfigurations-reader
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - contentconfigurations
        verbs:
          - get
          - watch
          - list
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: read-contentconfigurations-global
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: contentconfigurations-reader
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-portal
        namespace: NAMESPACE
  5: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-portal
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME-portal
      type: ClusterIP
